### PR TITLE
ticket #3225: Open several tabs with chrome makes the UI freeze fixed

### PR DESCRIPTION
**Pull request by nendhruv**
Ticket [#3225](http://trac.buildbot.net/ticket/3225)

### CODE REVIEW

File: www/base/src/app/common/services/mq/mq.service.coffee [hunk 1]
```diff
@@ -1,102 +1,132 @@
-class MqService extends Factory('common')
-    constructor: ($http, $rootScope, $q) ->
-        # private variables
-        match = (matcher, value) ->
-            # ultra simple matcher used to route event back to the original subscriber
-            matcher = new RegExp("^"+matcher.replace(/\*/g, "[^/]+") + "$")
-            return matcher.test(value)
+    class MqService extends Factory('common')
```
Comment (tardyp) [inline www/base/src/app/common/services/mq/mq.service.coffee]: why additional indent?

File: www/base/src/app/common/services/mq/mq.service.coffee [hunk 2]
```diff
@@ -1,102 +1,132 @@
...
-        listeners = {}
-        eventsource = null
-        cid = null
-        basepath = null
-        deferred = null
-        lostConnection = false
-        self =
-            # tested internal api
-            _match: match
+            if typeof document.hidden != 'undefined'
```
Comment (tardyp) [inline www/base/src/app/common/services/mq/mq.service.coffee]: this is not a standard api, no need for vendor prefix

**Comment by tardyp** [comment]
due to the reindent, it is very hard to review. please fix.

**Comment by tardyp** [comment]
Also, this needs unit tests

**Comment by djmitche** [comment]
What's the overall approach to solving this problem?  Generally a commit message should include some description of what the patch introduces, rather than just what it eliminates (so, "use foobars to improve frobnication" rather than "fix frobnication").  That helps readers figure out what to look for.

**Comment by nendhruv** [comment]
looking into all of this, will generate a new pull request.

**Comment by sa2ajj** [comment]
@nendhruv why?  Will not `git rebase -i ...` help you to do the job?

**Comment by tardyp** [comment]
as this is old mqservice based, better close this

**Pull request closed without merging**